server/src/internalClusterTest/java/org/elasticsearch/cluster/settings/ClusterSettingsIT.java

```diff
@@ -287,6 +289,145 @@ public void testUpdateSettings() {
+        restartNodesOnBrokenClusterState(ClusterState.builder(state).metadata(brokenMeta));
+        ensureGreen(); // wait for state recovery
+        state = client().admin().cluster().prepareState().get().getState();
+        assertTrue(state.getMetadata().persistentSettings().getAsBoolean("archived.this.is.unknown", false));
+
+        // cannot remove read only block due to archived settings
+        final IllegalArgumentException e1 =
+            expectThrows(
+                IllegalArgumentException.class,
+                () -> {
+                    Settings.Builder builder = Settings.builder();
+                    if (readOnly) {
+                        builder.put(Metadata.SETTING_READ_ONLY_SETTING.getKey(), "false");
+                    }
+                    if (readOnlyAllowDelete) {
+                        builder.put(Metadata.SETTING_READ_ONLY_ALLOW_DELETE_SETTING.getKey(), "false");
+                    }
+                    assertAcked(client().admin().cluster().prepareUpdateSettings()
+                        .setPersistentSettings(builder).setTransientSettings(builder).get());
+                });
+        assertTrue(e1.getMessage().contains("unknown setting [archived.this.is.unknown]"));
+
+        // fail to clear archived settings with non-archived settings
+        final ClusterBlockException e2 =
+            expectThrows(
+                ClusterBlockException.class,
+                () -> assertAcked(client().admin().cluster().prepareUpdateSettings()
+                        .setPersistentSettings(Settings.builder().putNull("cluster.routing.allocation.enable"))
+                        .setTransientSettings(Settings.builder().putNull("archived.*")).get()));
+        if (readOnly) {
+            assertTrue(e2.getMessage().contains("cluster read-only (api)"));
+        }
+        if (readOnlyAllowDelete) {
+            assertTrue(e2.getMessage().contains("cluster read-only / allow delete (api)"));
+        }
+
+        // fail to clear archived settings due to cluster read only block
+        final ClusterBlockException e3 =
+            expectThrows(
+                ClusterBlockException.class,
+                () -> assertAcked(client().admin().cluster().prepareUpdateSettings()
+                    .setPersistentSettings(Settings.builder().putNull("archived.*")).get()));
+        if (readOnly) {
+            assertTrue(e3.getMessage().contains("cluster read-only (api)"));
+        }
+        if (readOnlyAllowDelete) {
+            assertTrue(e3.getMessage().contains("cluster read-only / allow delete (api)"));
+        }
+
+        // fail to clear archived settings with adding cluster block
+        final ClusterBlockException e4 =
+            expectThrows(
+                ClusterBlockException.class,
+                () -> {
+                    Settings.Builder builder = Settings.builder().putNull("archived.*");
+                    if (readOnly) {
```

Comment:
I think here we should set these settings randomly, not just setting the one we care about to `true`.